src/pages/organize/[orgId]/journeys/[journeyId]/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { GetServerSideProps } from 'next';
+import Head from 'next/head';
+
+import AllJourneyInstancesLayout from 'layout/organize/AllJourneyInstancesLayout';
+import { ColumnNames } from 'components/journeys/JourneyInstancesDataTable/getColumns';
+import getOrg from 'fetching/getOrg';
+import JourneyInstancesDataTable from 'components/journeys/JourneyInstancesDataTable';
+import { PageWithLayout } from 'types';
+import { scaffold } from 'utils/next';
+import { TagMetadata } from 'pages/api/organize/[orgId]/journeys/[journeyId]/getTagMetadata';
+import ZetkinQuery from 'components/ZetkinQuery';
+import { journeyInstancesResource, journeyResource } from 'api/journeys';
+import { ZetkinJourney, ZetkinJourneyInstance } from 'types/zetkin';
+
+const scaffoldOptions = {
+  authLevelRequired: 2,
+  localeScope: ['layout.organize', 'misc.breadcrumbs', 'pages.organizeJourney'],
+};
+
+export const getServerSideProps: GetServerSideProps = scaffold(async (ctx) => {
+  const { orgId, journeyId } = ctx.params!;
+
+  await ctx.queryClient.prefetchQuery(
+    ['org', orgId],
+    getOrg(orgId as string, ctx.apiFetch)
+  );
+  const orgState = ctx.queryClient.getQueryState(['org', orgId]);
+
+  const { state: journeyQueryState } = await journeyResource(
+    orgId as string,
+    journeyId as string
+  ).prefetch(ctx);
+
+  if (
+    orgState?.status === 'success' &&
+    journeyQueryState?.status === 'success'
+  ) {
+    return {
+      props: {
+        journeyId,
+        orgId,
+      },
+    };
+  } else {
+    return {
+      notFound: true,
+    };
+  }
+}, scaffoldOptions);
+
+type JourneyInstancesOverviewPageProps = {
+  journeyId: string;
+  orgId: string;
+};
+
+const JourneyInstancesOverviewPage: PageWithLayout<
+  JourneyInstancesOverviewPageProps
+> = ({ orgId, journeyId }) => {
+  const journeyQuery = journeyResource(orgId, journeyId).useQuery();
+  const journeyInstancesQuery = journeyInstancesResource(
+    orgId,
+    journeyId
+  ).useQuery();
+  const journey = journeyQuery.data as ZetkinJourney;
+
+  return (
+    <>
+      <Head>
+        <title>{journey.plural_name}</title>
+      </Head>
+      <ZetkinQuery queries={{ journeyInstancesQuery }}>
+        <JourneyInstancesDataTable
+          {...{
+            journey,
+            ...(journeyInstancesQuery.data as {
+              columnNames: ColumnNames;
+              journeyInstances: ZetkinJourneyInstance[];
+              tagMetadata: TagMetadata;
+            }),
+          }}
```

Comment:
I find it difficult to read these. Is there any reason we can't do this?

```tsx
<JourneyInstancesDataTable
  journey={ journey }
  columnNames={ journeyInstancesQuery.columnNames }
  journeyInstances={ journeyInstancesQuery.journeyInstances }
  tagMetaData={ journeyInstancesQuery.tagMetaData }
/>
```

To me it's much easier to read and I imagine it's even more so for someone with less understanding of the code, or even React, as is the case with some volunteers.

Reply:
I've factored out types into an interface to make it easier